GPU/Vulkan/FramebufferVulkan.cpp

```diff
@@ -52,45 +54,224 @@
+
+	std::string fs_errors, vs_errors;
+	fsBasicTex_ = CompileShaderModule(vulkan_, VK_SHADER_STAGE_FRAGMENT_BIT, tex_fs, &fs_errors);
+	vsBasicTex_ = CompileShaderModule(vulkan_, VK_SHADER_STAGE_VERTEX_BIT, tex_vs, &vs_errors);
+	assert(fsBasicTex_ != VK_NULL_HANDLE);
+	assert(vsBasicTex_ != VK_NULL_HANDLE);
+
+	pipelineBasicTex_ = vulkan2D_.GetPipeline(pipelineCache2D_, rpClearColorClearDepth_, vsBasicTex_, fsBasicTex_);
+
+	VkSamplerCreateInfo samp = { VK_STRUCTURE_TYPE_SAMPLER_CREATE_INFO };
+	samp.addressModeU = VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_EDGE;
+	samp.addressModeV = VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_EDGE;
+	samp.addressModeW = VK_SAMPLER_ADDRESS_MODE_CLAMP_TO_EDGE;
+	samp.magFilter = VK_FILTER_NEAREST;
+	samp.minFilter = VK_FILTER_NEAREST;
+	VkResult res = vkCreateSampler(vulkan_->GetDevice(), &samp, nullptr, &nearestSampler_);
+	assert(res == VK_SUCCESS);
+	samp.magFilter = VK_FILTER_LINEAR;
+	samp.minFilter = VK_FILTER_LINEAR;
+	res = vkCreateSampler(vulkan_->GetDevice(), &samp, nullptr, &linearSampler_);
+	assert(res == VK_SUCCESS);
+}
+
+FramebufferManagerVulkan::~FramebufferManagerVulkan() {
+	vulkan_->Delete().QueueDeleteRenderPass(rpLoadColorLoadDepth_);
+	vulkan_->Delete().QueueDeleteRenderPass(rpClearColorLoadDepth_);
+	vulkan_->Delete().QueueDeleteRenderPass(rpClearColorClearDepth_);
+	vulkan_->Delete().QueueDeleteRenderPass(rpLoadColorClearDepth_);
+
+	for (int i = 0; i < 2; i++) {
+		if (frameData_[i].numCommandBuffers_ > 0) {
+			vkFreeCommandBuffers(vulkan_->GetDevice(), frameData_[i].cmdPool_, frameData_[i].numCommandBuffers_, frameData_[i].commandBuffers_);
+		}
+		vkDestroyCommandPool(vulkan_->GetDevice(), frameData_[i].cmdPool_, nullptr);
+		frameData_[i].push_->Destroy(vulkan_);
+		delete frameData_[i].push_;
 	}
-	VkClearAttachment clear[2];
-	memset(clear, 0, sizeof(clear));
-	clear[0].aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
-	clear[1].aspectMask = VK_IMAGE_ASPECT_DEPTH_BIT | VK_IMAGE_ASPECT_STENCIL_BIT;
-	VkClearRect rc;
-	rc.baseArrayLayer = 0;
-	rc.layerCount = 1;
-	rc.rect.offset.x = 0;
-	rc.rect.offset.y = 0;
-	rc.rect.extent.width = currentRenderVfb_->bufferWidth;
-	rc.rect.extent.height = currentRenderVfb_->bufferHeight;
-	vkCmdClearAttachments(curCmd_, 2, clear, 1, &rc);
+	delete drawPixelsTex_;
+
+	vulkan_->Delete().QueueDeleteShaderModule(fsBasicTex_);
+	vulkan_->Delete().QueueDeleteShaderModule(vsBasicTex_);
+
+	vulkan_->Delete().QueueDeleteSampler(linearSampler_);
+	vulkan_->Delete().QueueDeleteSampler(nearestSampler_);
+	vulkan_->Delete().QueueDeletePipeline(pipelineBasicTex_);
+	if (pipelinePostShader_ != VK_NULL_HANDLE)
+		vulkan_->Delete().QueueDeletePipeline(pipelinePostShader_);
+	vulkan_->Delete().QueueDeletePipelineCache(pipelineCache2D_);
 }
 
-void FramebufferManagerVulkan::DisableState() {
+void FramebufferManagerVulkan::NotifyClear(bool clearColor, bool clearAlpha, bool clearDepth, uint32_t color, float depth) {
+	if (!this->useBufferedRendering_) {
```

Comment:
I think you didn't mean to include `this->` (I like being explicit sometimes but we're usually not..., better to be consistent...)

-[Unknown]